docs/src/man/joins1.md

```diff
@@ -0,0 +1,609 @@
+# Joining DataFrames
+
+"I have two different data sets in Julia but I am not sure how to use them. What 
+criteria should I consider? What are the different ways I can join these data sets?"
+
+Sound familiar? You may have come across this question plenty of times on online discussion
+forums. Working with one dataframe is fairly straightforward but things become challenging when
+we have data spread across two or more data frames. This is where the concept of `joins` come in.
+
+`Joins` is another very common and important operation that arises in the world of tabulated data. 
+A join across two DataFrames is the action of combining the two datasets based on shared column values 
+that exist across the two tables. We call this column (or columns) the `key`. So, each record from the 
+first table is matched to a record in the second table — as long as the records’ values are the same. 
+
+I’m sure you’re quite familiar with e-commerce sites like Amazon and Flipkart these days. We are 
+bombarded by their advertisements when we’re visiting non-related websites – that’s the power of 
+targeted marketing! Previously, we have used German Credit Card dataset. Now let's say you are working
+on a Classification problem but we have not target column in our dataset. We are given two data frames – 
+one which contains data about customer and the other that has taget information. We will use these dataframes 
+to understand how the different types of joins work using `DataFrames.jl`.
+
+For example, suppose that we have the following two datasets:
+
+```jldoctest dataframe
+julia> using DataFrames
+
+julia> using CSV
+
+julia> german_ref = CSV.read((joinpath(dirname(pathof(DataFrames)),
+                                    "..", "docs", "src", "assets", "german.csv")),
+                                    DataFrame)
+1000×10 DataFrame
+  Row │ id     Age    Sex     Job    Housing  Saving accounts  Checking accoun ⋯
+      │ Int64  Int64  String  Int64  String   String           String          ⋯
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │     0     67  male        2  own      NA               little          ⋯
```

Comment:
what does `NA` mean? Is this a missing value? then we should properly read it in (the same for the other PR)